x-pack/plugins/security_solution/public/common/components/import_data_modal/utils.ts

```diff
@@ -33,8 +33,11 @@ export const formatError = (
 const mapErrorMessageToUserMessage = (
   actionConnectorsErrors: Array<ImportRulesResponseError | ImportResponseError>
 ) => {
-  return actionConnectorsErrors.map((connectorError) => {
-    const { error } = connectorError;
+  let concatenatedActionIds: string = '';
+  const mappedErrors = actionConnectorsErrors.map((connectorError) => {
+    const { id, error } = connectorError as ImportResponseError;
```

Comment:
Why isn't it possible to just calculate the number of errors? Can we get more than one error per connector?

Generally speaking `ImportRulesResponseError | ImportResponseError` only differs by `id` and `rule_id` which come from [BulkError](https://github.com/elastic/kibana/blob/main/x-pack/plugins/security_solution/server/lib/detection_engine/routes/utils.ts#L24-L25) (just the fact `BulkError` allows to omit both `id` and `rule_id` which is not supposed to happen I expect but it's another story). If we still support `rule_id` (most probably for backwards compatibility) can we have a situation when it's in use and this code is gonna fail? If it's not a case so then we should remove `rule_id` so `as ImportResponseError` type casting won't be necessary.

Reply:
`Why isn't it possible to just calculate the number of errors? Can we get more than one error per connector?` 

-Yes true.
Forex. if the user has rule to import in one file, and this rule has 3 connectors.

For instance, if 2 of them have missing connectors, what will happen as per the Old workflow=> One error message will be generated with the count of the failing connectors, and one error object will be pushed in the array.
` 2 connectors are missing. Connector ids missing are: X, Y`


And just to clarify the `id` here is the action-connectors `id` not the `rule_id`

Reply:
Hey @maximpn ! `rule_id` was initially created to act as a signature id so that a rule could be recognized, say, across platforms. This allows sigma, snort, suricata etc to identify a rule with a singular "signature id" that is never regenerated over time. So the concept of `rule_id` should hopefully stick around the solution unless platform has come up with a way for it to be supported more globally within a saved object. This comes in very handy on import when a user could be importing a rule they've ported over from elsewhere and want to maintain it's reference to the signature id. I don't think it would be in our interest to remove it's use. Like @WafaaNasr pointed out, the `id` she references is to the connector, the `rule_id` points back to the rule that had an action pointing to said connector so the user knows which rule was the source of the error.

If there's a worry that `id` would not exist in `connectorError` because it ends up being `ImportRulesResponseError` vs `ImportResponseError` maybe we can add a check for `id != null`.

If there's larger refactoring you think is necessary (this route is now a bit dated and could benefit from refactoring as it's complexity has grown a lot), feel free to create a ticket and we can keep discussing what the response should look like and how `rule_id` fits into the import use case.

Reply:
Hey @yctercero, thank you for the explanation! My concern is misleading types as `BulkError` at backend and  `ImportRulesResponseError` at frontend don't look related to the connectors. `ImportResponseError` looks too generic (by name). According to what I see in the code it's pretty easy to use some tailored for connectors error type which `action_connectors_errors` will use.

Absence of such a type leads to reusing `id` and `rule_id` fields to store serialized arrays in a case of missing connectors. I'm afraid it doesn't match with SO importer errors so that importer seems to returning an error per connector. Merging it now and creating a ticket looks like adding a technical debt while we already have it a lot.

Reply:
There's definitely refactoring needed overall on import. It used to just be rules and now is exceptions, actions and connectors. With the added comments, we can certainly look to simplify and refactor this flow.

Reply:
I understand the concern with creating technical debt. As we are far into the BCs, adding scope in this bug fix can also open us up to further bugs. Our team is working on documenting this route, the technical debt and suggested updates. I can be sure to make reference back to your concerns to ensure they're addressed in any upcoming refactors.

Reply:
Here we go https://github.com/elastic/kibana/issues/152648

Reply:
Thank you @maximpn !